core/src/main/java/org/opensearch/sql/data/type/ExprType.java

```diff
@@ -64,4 +64,18 @@ default List<ExprType> getParent() {
   default String legacyTypeName() {
     return typeName();
   }
+
+  /**
+   * Perform field name conversion if needed before inserting it into a search query.
+   */
+  default String convertFieldForSearchQuery(String fieldName) {
+    return fieldName;
+  }
+
+  /**
+   * Perform value conversion if needed before inserting it into a search query.
+   */
+  default Object convertValueForSearchQuery(ExprValue value) {
+    return value.value();
+  }
```

Comment:
Unfortunately, where it is used, `ExprType` is referenced. I'd like to avoid excessive refactoring there.
https://github.com/Bit-Quill/opensearch-project-sql/blob/6c3744ec81931567d50f47075cd45b3e8906997c/opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/LikeQuery.java#L16-L19
Any ideas how to do it gracefully?